src/App.tsx

```diff
@@ -1,26 +1,222 @@
-import React from 'react';
+/* eslint-disable */
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, createTodos, deleteTodo, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
+  const [error, setError] = useState<string | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isTitle, setTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoadingTodo, setIsLoadingTodo] = useState<number[]>([]);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterStatus) {
+      case 'active':
+        return !todo.completed;
+      case 'completed':
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  useEffect(() => {
+    if (USER_ID) {
+      handleRequest();
+    }
+
+    setTimeout(() => {
+      setError(''); // Очищаємо помилку після схову повідомлення
+    }, 3000);
+  }, []);
+
+  // Відправляє запит на сервер для отримання списку (todos).
+  const handleRequest = async () => {
+    try {
+      setLoading(true);
+      const allTodo = await getTodos();
+
+      setTodos(allTodo);
+      setError(null); // При успішному завершенні запиту помилка схована
+    } catch (errors) {
+      // Обробка помилок, якщо вони виникають під час відправлення запиту
+      setError('Unable to load todos');
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  // додавання фокусу на input
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [todos]);
+
+  // Відповідає за обробку події форми.
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (isTitle.trim() === '') {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    try {
+      setTempTodo({
+        id: 0, // create a todo with id: 0
+        title: isTitle.trim(),
+        completed: false,
+        userId: USER_ID,
+      });
+
+      // викликаємо функцію createTodos, яка відправляє запит на сервер та створює новий todo
+      const newTodo = await createTodos({
+        title: isTitle.trim(),
+        completed: false,
+        userId: USER_ID,
+      });
+
+      //  додає новий елемент до масиву todos та оновлює його стан
+      setTodos(prevTodos => [...prevTodos, newTodo]);
+      setTitle('');
+    } catch (errors) {
+      setError('Unable to add a todo');
+    } finally {
+      setTempTodo(null); // Сховати tempTodo
+    }
+  };
+
+  const deleteTodos = async (userId: number) => {
```

Comment:
```suggestion
  const deleteTodo = async (userId: number) => {
```
it deletes one todo
you can `import { deleteTodo as deleteTodoApi } from 'path/example'` to avoid variable name duplication